File: frontend/src/api/irDocument.ts
import api from './index'

export interface IRDocument {
  id: number
  requirementId: number
  content: string
  status: 'DRAFT' | 'VALID' | 'INVALID' | 'LOCKED'
  validationErrors: string | null
  createdBy: string
  updatedBy: string
  createdAt: string
  updatedAt: string
}

export const irDocumentApi = {
  getByRequirementId: (requirementId: number) => {
    return api.get<IRDocument>(`/ir-documents/requirement/${requirementId}`)
  },

  create: (requirementId: number, content: string) => {
    return api.post<IRDocument>('/ir-documents', content, {
      params: { requirementId },
      headers: { 'Content-Type': 'application/json' }
    })
  },

  update: (id: number, content: string) => {
    return api.put<IRDocument>(`/ir-documents/${id}`, content, {
      headers: { 'Content-Type': 'application/json' }
    })
  },

  validate: (id: number) => {
    return api.post<IRDocument>(`/ir-documents/${id}/validate`)
  },

  delete: (id: number) => {
    return api.delete(`/ir-documents/${id}`)
  }
}

File: backend/src/main/java/com/aigen/studio/entity/IRDocument.java
package com.aigen.studio.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "ir_documents")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class IRDocument {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "requirement_id", nullable = false)
    private Long requirementId;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IRStatus status;

    @Column(columnDefinition = "TEXT")
    private String validationErrors;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "updated_by")
    private String updatedBy;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum IRStatus {
        DRAFT, VALID, INVALID, LOCKED
    }
}